src/tequila/circuit/compiler.py

```diff
@@ -1248,12 +1248,14 @@ def compile_generalized_rotation_gate(gate, compile_exponential_pauli: bool = Fa
     -------
 
     """
-    if not hasattr(gate, "generator"):
+    if gate.generator is None or gate.name.lower() in ['phase']:
```

Comment:
will currently also compile standard rotations (formally correct, but the trotter_compile step will lead to them being compiled into Rz rotations with basis change, a lot of interfaces/hardware support Rx and Ry rotations though).

Try to excude them also ( think you can just add rz, ry, rz in the list with phase)

Reply:
This is also what causes the erratic behaviour in  the UnaryStatePrep test. Excluding the primitive rotations from being compiled fixes it

Reply:
I think they will stay as Rx and Ry due to [L372-L375](https://github.com/aspuru-guzik-group/tequila/blob/master/src/tequila/circuit/gates.py#L372).
Since it fixes the failing test, i exclude them.